bundles/org.openhab.binding.icloud/src/main/java/org/openhab/binding/icloud/internal/utilities/CustomCookieStore.java

```diff
@@ -0,0 +1,111 @@
+/**
+ * Copyright (c) 2010-2022 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.icloud.internal.utilities;
+
+import java.net.CookieManager;
+import java.net.CookieStore;
+import java.net.HttpCookie;
+import java.net.URI;
+import java.util.List;
+
+import org.openhab.core.storage.Storage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+/**
+ *
+ * This class implements a customized {@link CookieStore}. Its purpose is to add hyphens at the beginning and end of
+ * each cookie value which is required by Apple iCloud API.
+ *
+ *
+ * @author Simon Spielmann Initial contribution
+ */
+public class CustomCookieStore implements CookieStore {
```

Comment:
Please annotate with `@NonNullByDefault`.

Reply:
This produces a lot of error because of unmatched null-typed from supertype. Tried a few variants, but couldn't get it compile. Do you have a hint or is it okay to have this derived class without this annotation?

Reply:
All new classes should be annotated. I'll see if I can find time to have a look in the evening, if not @lsiepel will not have checked already. :)

Reply:
Sorry, I'm unable to get the annotation to work for this class. Can you help out? Or merge as is?

Reply:
Unfortunately I was not able to sort it out tonight. Let's focus on the other comments first, and this one we can let slip if we won't manage in time.